src/main/java/com/app/folioman/mfschemes/service/NewlyInsertedSchemesListener.java

```diff
@@ -17,22 +15,18 @@ public class NewlyInsertedSchemesListener {
     private static final Logger log = LoggerFactory.getLogger(NewlyInsertedSchemesListener.class);
 
     private final MFNavService mfNavService;
-    private final TaskExecutor taskExecutor;
 
-    public NewlyInsertedSchemesListener(
-            MFNavService mfNavService, @Qualifier("taskExecutor") TaskExecutor taskExecutor) {
+    public NewlyInsertedSchemesListener(MFNavService mfNavService) {
         this.mfNavService = mfNavService;
-        this.taskExecutor = taskExecutor;
     }
 
     @ApplicationModuleListener
+    @EventListener
```

Comment:
@coderabbitai, what changes should be done to use `@ApplicationModuleListener` and completely avoid using `@EventListener` in main src code to make integration tests pass